usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,40 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+START -> The state of the instance when this code is executed
+         This code is executed by a timer
+
+    START   LICENSE CHANGED TO     CACHE   ACTION
+1   BYOS-1  PAYG                   No      Register - Update Infra
+2   BYOS-1  PAYG                   BYOS    Register - Update Infra
+3   BYOS-1  BYOS-1                 No      Do nothing
+4   BYOS-1  BYOS-1                 BYOS    Do nothing
+5   BYOS-1  BYOS-2                 No      Do nothing
+6   BYOS-1  BYOS-2                 BYOS    Do nothing
+
+7   BYOS-2  PAYG                   No      Do nothing
+8   BYOS-2  PAYG                   BYOS    Re-register - Update Infra
+9   BYOS-2  BYOS-1                 No      De-register - Update Infra
+10  BYOS-2  BYOS-1                 BYOS    De-register - Update Infra
+11  BYOS-3  PAYG                   No      No de-register from SCC, register with update infrastructure
+12  BYOS-3  PAYG                   BYOS    No de-register from SCC, register with update infrastructure
+
+13  PAYG    BYOS-1                 No      De-register - Update Infra
+14  PAYG    BYOS-1                 PAYG    De-register - Update Infra
+15  PAYG    BYOS-2,3               No      Do nothing
+16  PAYG    BYOS-2,3               PAYG    Do nothing
```

Comment:
Here too, the customer cannot change the license state to `BYOS-2` or `BYOS-3`. Since we have trouble with this I think the column should be renamed from `LICENSE CHANGED TO` to `Reg State at runtime` this would indicate that we start with a given state and then detect the state during runtime of the code. And the state is set by the user as a combination of running AHB commands and registration commands. This should get us out of the quandary we are in. Let's try that and see what that looks like.